presto-parser/src/test/java/com/facebook/presto/sql/parser/TestSqlParser.java

```diff
@@ -1039,13 +1057,142 @@ public void testSelectWithGroupBy()
+                                        new SingleColumn(
+                                                new FunctionCall(
+                                                        QualifiedName.of("row_number"),
+                                                        Optional.of(
+                                                                new WindowInline(
+                                                                        new WindowSpecification(
+                                                                                Optional.empty(),
+                                                                                ImmutableList.of(),
+                                                                                ImmutableList.of(new SortItem(new QualifiedNameReference(QualifiedName.of("b")), SortItem.Ordering.ASCENDING, SortItem.NullOrdering.FIRST)),
+                                                                                Optional.empty()))),
+                                                        false,
+                                                        ImmutableList.of()))),
+                                Optional.of(new Table(QualifiedName.of("c"))),
+                                Optional.empty(),
+                                Optional.empty(),
+                                Optional.empty(),
+                                ImmutableList.of(),
+                                ImmutableList.of(),
+                                Optional.empty()),
+                        ImmutableList.of(),
+                        Optional.empty(),
+                        Optional.empty()));
+
+        assertStatement("SELECT a, row_number() OVER b FROM c",
+                new Query(
+                        Optional.empty(),
+                        new QuerySpecification(
+                                selectList(
+                                        new SingleColumn(new QualifiedNameReference(QualifiedName.of("a"))),
+                                        new SingleColumn(
+                                                new FunctionCall(
+                                                        QualifiedName.of("row_number"),
+                                                        Optional.of(new WindowName("b")),
+                                                        false,
+                                                        ImmutableList.of()))),
+                                Optional.of(new Table(QualifiedName.of("c"))),
+                                Optional.empty(),
+                                Optional.empty(),
+                                Optional.empty(),
+                                ImmutableList.of(),
+                                ImmutableList.of(),
+                                Optional.empty()),
+                        ImmutableList.of(),
+                        Optional.empty(),
+                        Optional.empty()));
+    }
```

Comment:
It lacks tests of full specification (with `partition by` clause). What's more, there should be  tests covering multiple functions parsed properly into a list of functions.